api/apps/api/src/migrations/api/1694192071502-AddScenarioToCostSurfaceAPIRelationship.ts

```diff
@@ -0,0 +1,16 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class AddScenarioToCostSurfaceAPIRelationship1694192071502
+  implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`
+        ALTER TABLE scenarios ADD COLUMN cost_surface_id uuid NOT NULL REFERENCES cost_surfaces(id);
```

Comment:
I think we'll have to make this nullable here, then within this same migration file:

- create as many default cost surfaces as there are project in a given MaPP instance, each linked to its project
- create as many cost surfaces as there are scenarios and link each of these to its scenario
- _copy_ all the `scenarios_pu_cost_data` over from each scenario's set of `scenarios_pu_cost_data` to the relevant cost surface data
- and finally make the `scenarios.cost_surface_id` fk not nullable

Alternatively, go for any of the Python or bash/psql based data migration paths you've mentioned, in which case this migration should "just" add the new column as nullable, then data migrations need to be done manually (do we _really want_ to do this, though?) and only then deploy a new "release" that includes a migration that makes the new column not null.

Reply:
Hullo @hotzevzl! We discussed a couple of ways to manage this. 
Here is a ugly py script that should be able to handle this and just reflect the database changes with the above migration (adding required IF NOT EXISTS statements...)

https://github.com/Vizzuality/marxan-cloud/pull/1501

Let's discuss it later